src/scenic/core/regions.py

```diff
@@ -281,883 +306,2816 @@ def __hash__(self):
+
+    By default the unit sphere centered at the origin and aligned with the axes is used.
+
+    Args:
+        name: An optional name to help with debugging.
+        dimensions: An optional 3-tuple, with the values representing width, length, height respectively.
+          The mesh will be scaled such that the bounding box for the mesh has these dimensions.
+        position: An optional position, which determines where the center of the region will be.
+        rotation: An optional Orientation object which determines the rotation of the object in space.
+        orientation: An optional vector field describing the preferred orientation at every point in
+          the region.
+        tolerance: Tolerance for collision computations.
+        engine: Which engine to use for mesh operations. Either "blender" or "scad".
+    """
+    def __init__(self, name=None, dimensions=None, position=None, rotation=None, orientation=None,
+                 tolerance=1e-8, onDirection=None, engine="blender"):
+        sphere_mesh = trimesh.creation.icosphere(radius=1)
+        super().__init__(mesh=sphere_mesh, name=name, position=position, rotation=rotation, dimensions=dimensions, \
+            orientation=orientation, tolerance=tolerance, onDirection=onDirection, engine=engine)
+
+    @cached_property
+    def isConvex(self):
+        return True
+
+    def sampleGiven(self, value):
+        return SpheroidRegion(name=self.name, dimensions=value[self.dimensions],
+            position=value[self.position], rotation=value[self.rotation], \
+            orientation=self.orientation, tolerance=self.tolerance, engine=self.engine)
+
+class PolygonalFootprintRegion(Region):
+    """ Region that contains all points in a polygonal footprint, regardless of their z value. 
+
+    This region cannot be sampled from, as it has infinite height and therefore infinite volume.
+    
+    Args:
+        polygon: A ``MultiPolygon`` that defines the footprint of this region.
+        name: An optional name to help with debugging.
+    """
+    def __init__(self, polygon, name=None):
+        if not isinstance(polygon, MultiPolygon):
+            raise RuntimeError("'polygon' must be a shapely MultiPolygon")
+
+        self.polygon = polygon
+        super().__init__(name)
+        self._bounded_cache = None
+
+    def intersect(self, other, triedReversed=False):
+        """ Get a `Region` representing the intersection of this region with another.
+
+        This function handles intersection computation for `PolygonalFootprintRegion` with:
+        * `PolygonalFootprintRegion`
+        * `PolygonalRegion`
+        """
+        if isinstance(other, PolygonalFootprintRegion):
+            # Other region is a PolygonalFootprintRegion, so we can just intersect the base polygons
+            # and take the footprint of the result, if it isn't empty.
+            new_poly = PolygonalRegion(polygon=self.polygon).intersect(PolygonalRegion(polygon=other.polygon))
```

Comment:
Wouldn't it be easier to just do `new_poly = self.polygon & other.polygon`? I guess this would break the `PolygonalFootprintRegion` initializer if the intersection returns a `Polygon`, but really we should handle that case if we expect people to use this region type.

Reply:
I don't think people should ever create this region manually, since they can always just grab it from the `.footprint` parameter, and it's mostly used by Scenic internally.

Reply:
Still, I went ahead and took that step out since it'll hopefully be faster.